services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/client_api.py

```diff
@@ -246,6 +247,10 @@ async def service_push_output_ports(
         response = await self._client.post(
             url, json=port_keys, timeout=self._save_restore_timeout
         )
+        if response.status_code == status.HTTP_404_NOT_FOUND:
+            # check if the error of the dict is as expected an raise to skip!
+            if "node_not_found" in response.text:
```

Comment:
Please think carefully about error handling. Checking for a string in a text classify is very unreliable and difficult to maintain. 

We had a long discussion about how e.g. pydantic deals with error handling.  

Some tips: 
- Transform ``NodeNotFound`` into http-error class but using structured json-response SEE e.g. http error handler in ``services/director-v2/src/simcore_service_director_v2/api/errors/http_error.py``
- Create a code system to classify errors (We even extended pydantic mechanism in )